### ISSUE TITLE
Guru navigation should support build tags

### ISSUE THREAD

**Issue by krasi-georgiev** [joefitzgerald/go-plus#603]
Atom 1.15.0

this happens only for some packages.
A quick google search show that it has something to do with the 
//+build tags

$ /usr/local/Cellar/go/1.8/libexec/bin/go version
go version go1.8 darwin/amd64

$ /usr/local/Cellar/go/1.8/libexec/bin/go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/krasimir"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.8/libexec"
GOTOOLDIR="/usr/local/Cellar/go/1.8/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PHP DebugPHP ConsoleSettings
 go-plus6 deprecations

* [x] Have you tried launching `atom .` from the terminal in your project's directory?
* [x] Have you verified the output from `go env` is correct? If it is, please include the output in this issue.
* [x] Have you updated Atom to the latest version?
* [ ] Have you tried using [Atom Beta](https://atom.io/beta), which can be run side-by-side with Atom Stable? Is the behavior the same, or different?
No, Don't think it is relevant.
* [x] Have you updated your Atom packages to the latest versions?
* [x] Have you read the [FAQ](https://github.com/joefitzgerald/go-plus/wiki/FAQ)?
* [x] Have you searched [the issues](https://github.com/joefitzgerald/go-plus/issues?utf8=%E2%9C%93&q=is%3Aissue) to see if others are experiencing the same issue?

### Output from `atom -v && apm -v`

Atom    : 1.15.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
apm  1.15.3
npm  3.10.5
node 4.4.5 x64
python 2.7.10
git 2.10.1

### Output From `go env`

GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/krasimir/projects/vipconsult/golang"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"

### Steps to Reproduce
https://screencast.com/t/LC2xKqlc

### Expected Behavior

to navigate to the correct package
### Actual Behavior
error

**Comment by krasi-georgiev** [comment]
I think a possible fix is to simply allow overriding the compilation target.

for example in my use case I change and navigate the code on OSX/darwin, but the compilation happens in a docker container which is debian/linux

**Comment by krasi-georgiev** [comment]
now I hit another issue when trying to compile a darwin package I need change the $GOOS every time

would it make sense to allow setting custom env in go-plus instead of the default ~/.bash_profile ?

**Comment by zmb3** [comment]
I wouldn't recommend using your text editor to cross compile. Go plus runs
tests so needs to be building artifacts for your host OS.

Whenever I need to cross compile I simply run a Go build from the command
line.

**Comment by krasi-georgiev** [comment]
my issue is code navigation CTRL+click to go to function definition.

**Comment by zmb3** [comment]
Right, but setting `GOOS` affects everything. For example:

- Go-Plus will be cross compiling your builds, which is often slower and may pollute your `GOPATH`
- Tests won't run since you would be compiling for another OS

The real issue here is that we need to pass `-tags` to the guru tool.  Right now there's no way to configure this, but we've started work on project-based config (#597) that would allow you to specify build tags.  I've updated the title of this issue to clarify.
